packages/react/src/Composite/Root/useCompositeRoot.ts

```diff
@@ -105,12 +96,19 @@ export function useCompositeRoot(params: UseCompositeRootParameters) {
         ref: mergedRef,
         onKeyDown(event) {
           const RELEVANT_KEYS = enableHomeAndEndKeys ? ALL_KEYS : ARROW_KEYS;
-
           if (!RELEVANT_KEYS.includes(event.key)) {
             return;
           }
 
-          const isRtl = textDirectionRef?.current === 'rtl';
+          const element = rootRef.current;
+          if (!element) {
+            return;
+          }
+
+          const isRtl =
+            textDirectionRef?.current == null
+              ? getTextDirection(element) === 'rtl'
```

Comment:
This function triggers a reflow, proof: https://gist.github.com/paulirish/5d52fb081b3570c81e3a?permalink_comment_id=5299127#gistcomment-5299127